### PR TITLE
[fixed] Make GHA workflow to retrieve latest promote-able SHA from master

### PR DESCRIPTION
**Pull request by swang392**
Relates to #76700

**Overview**: Wrote GHA to get the latest commit SHA. Another component of the script is pushing this SHA to the viable/strict branch, which I will test on pytorch/pytorch-canary.

Todo in the next PR: add comment explaining cron, replace package installation statements with txt file

**Test Plan:** Monitor github actions results to see if the SHA printed is correct by running GHA on pytorch/pytorch-canary. The successful test workflow is [here](https://github.com/pytorch/pytorch-canary/runs/6888486129?check_suite_focus=true).

### CODE REVIEW

File: .github/workflows/update-viablestrict.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: Update viable/strict
+
+on:
+  schedule:
+    - cron: 17,47 * * * *
```
Comment (janeyx99) [inline .github/workflows/update-viablestrict.yml:5]: Not urgent to address in this PR, but good to have a comment explaining the cron

File: .github/workflows/update-viablestrict.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  do_update_viablestrict:
+    runs-on: ubuntu-20.04
```
Comment (janeyx99) [inline .github/workflows/update-viablestrict.yml:14]: @seemethere is this a good runner for these? I think these are okay, but maybe one of the premium runners?

**Review by janeyx99 — APPROVED** [review]

File: .github/workflows/update-viablestrict.yml (hunk 2 above)
Comment (seemethere) [reply .github/workflows/update-viablestrict.yml:14]: Since this doesn't run very often I think these should be fine to use

**Comment by kit1980** [comment]
Please rebase on master to prevent asan (and maybe some other) test failures.

**Pull request closed without merging**